s3tests_boto3/functional/test_iam.py

```diff
@@ -2875,3 +2875,368 @@ def test_allow_deny_delete_object_version_tagging_iam_policy_others():
     _empty_versioned_bucket(s3_client_alt, bucket)
     response = s3_client_alt.delete_bucket(Bucket=bucket)
     eq(response['ResponseMetadata']['HTTPStatusCode'], 204)
+
+#47415
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Allow PutObject, GetObject, GetObjectTagging with source object is without tags')
+@attr(assertion='succeeds')
+@attr('user-policy')
+@attr('test_of_iam')
+def test_allow_copy_object_put_get_getobjtagging_empty_source_tags():
+    client = get_iam_client()
+    s3_client_iam = get_iam_s3client()
+    s3_client_alt = get_alt_client()
+    key_src = "iam1bukobj"
+    key_dest = "iam2bukobj"
+    # Create bucket, upload object for user1
+    bucket_src = get_new_bucket(client=s3_client_iam)
+    response = s3_client_iam.put_object(Bucket=bucket_src, Key=key_src, Body='bar')
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    bucket_dest = get_new_bucket(client=s3_client_alt)
+
+    # Apply Allow s3:GetObject, PutObject, GetObjectTagging on user1buck for user2
+    allow_get_put_get_tagging_policy = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Allow",
+             "Action": ["s3:PutObject", "s3:GetObject", "s3:GetObjectTagging"],
```

Comment:
Either you done need PutObject permission or you want to copy from bucket 1 to bucket 2 of user 1

Reply:
source and destination buckets are of same user1